packages/repository/src/decorators/model.decorator.ts

```diff
@@ -91,6 +92,16 @@ export function model(definition?: Partial<ModelDefinitionSyntax>) {
  * @returns {(target:any, key:string)}
  */
 export function property(definition?: Partial<PropertyDefinition>) {
+  if (
+    definition &&
+    (definition.type === Array || definition.type === 'array') &&
+    (definition as object).hasOwnProperty('itemType') &&
+    !definition.itemType
+  ) {
+    // this path is taken when cyclic dependency is detected
+    // in that case, a ModelResolver should be used instead
+    throw new Error('target model is undefined');
```

Comment:
I believe that error messages should remain as literal as possible, but in this case it might be more reasonable to suggest users to use a resolver since the condition that checks for this code path is pretty hard to meet any other way.

I think I'll make this change

Reply:
it's better when an error message can tell you how to fix the problem imo :)